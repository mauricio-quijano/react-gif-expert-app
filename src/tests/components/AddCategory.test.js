import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Tests on <AddCategory /> component', () => {
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories} />);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    });

    test('should display component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should not submit form', () => {
        wrapper.find('form').simulate('submit', {
            preventDefault() {},
        });
        expect(setCategories).not.toHaveBeenCalled();
    });

    test('should call setCategories and then clean text box', () => {
        const value = 'Any text';
        const input = wrapper.find('input');

        input.simulate('change', {
            target: {
                value,
            },
        });

        wrapper.find('form').simulate('submit', {
            preventDefault() {},
        });

        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
        expect(input.prop('value')).toBe('');
    });
});
