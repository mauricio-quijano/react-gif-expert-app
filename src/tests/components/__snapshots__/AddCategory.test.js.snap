// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <AddCategory /> component should display component correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
